Add YAML / python output for update_from_* subroutines in ndrop.F90

brigaudet opened the pull request:
Create YAML / python output files for the following subroutines within ndrop.F90/dropmixnuc:

1)  update_from_newcld.  Changes from temporal changes in cloud fraction.  Files are tagged by:  iact0 = if/then branch for no activation, conversion of cloudborne species to interstitial; iact1 = if/then branch for activation, loadaer and activate_modal are ultimately called; iact2 = no action taken.  Input / output are for given column and vertical level.
2)  update_from_cldn_profile.  Changes from vertical gradients of cloud fraction.  Files are tagged by:  iact0 = if/then branch for no activation, conversion of cloudborne species to interstitial; iact1 = if/then branch for activation, loadaer and activate_modal are ultimately called; iact2 = no action taken.  Input / output are for given column and vertical level = kk, but for some fields also for vertical level = kk+1.
3)  update_from_explmix.  Changes from explicit vertical mixing.  Input / output are for all vertical levels within given column.  Two time levels for raercol, raercol_cw arrays are output separately.

Commit:
Add YAML / python output for update_from_newcld, update_from_cldn_profile, and update_from_explmix subroutines within ndrop.F90

odiazib commented on components/eam/src/chemistry/yaml/cam_ndrop/update_from_cldn_profile_input_ts_1400_iact1.yaml:
```diff
@@ -0,0 +1,168 @@
+    ntot_amode: [         4]
+    nspec_amode: [         7,         4,         7,         3]
+# Data at [lat=   39.0553303768561      , lon=   262.904774388088      , k=          54 , time step=        1400 ] 
+mam4xx:
```
Does the info below correspond to a different condition? if so, please save it in a different yaml file.  I am not 100% sure if our code will handle a yaml file with multiple inputs sections. @jeff-cohere any thoughts? 

please review yaml files for `update_from_cldn_profile*`. There are multiple sections of mam4xx:

jeff-cohere replied:
You're right, @odiazib . These sets of input should be in separate files--Skywalker doesn't handle multiple ensembles in a single file, and I think it's simple if we keep it that way.

brigaudet replied:
Sorry -- there was a bug in the code to generate data for update_from_cldn_profile, it should have been printing only once per timestep.  The bug has now been corrected.

odiazib replied:
No worries. Thanks for fixing this issue.

odiazib commented on components/eam/src/chemistry/yaml/cam_ndrop/f90_yaml/update_from_newcld_end_yml.f90:
```diff
@@ -0,0 +1,41 @@
+
+     !start writing data
+     
+        call write_var(unit_output,'qcld',qcld)
```
Because inputs correspond to only one cell, you should save only one cell in the outputs. In this case :`yaml%lev_print`

brigaudet replied:
OK, this has been fixed.

odiazib commented on components/eam/src/chemistry/yaml/cam_ndrop/f90_yaml/update_from_cldn_profile_end_yml.f90:
```diff
@@ -0,0 +1,44 @@
+
+     !start writing data
+     
+        call write_var(unit_output,'raercol_nsav(yaml%lev_print,:)',raercol_nsav)
```
save only one cell in outputs : `raercol_nsav(yaml%lev_print,:)`

brigaudet replied:
This has been fixed.

Commit:
Correct bug in update_from_cldn_profile_beg_yml.f90, to only print once per timestep.
Regenerate data.

odiazib commented on components/eam/src/chemistry/yaml/cam_ndrop/f90_yaml/update_from_cldn_profile_beg_yml.f90:
```diff
@@ -0,0 +1,172 @@
+        ! for cs_col_in, values at both kk and kp1 are used for output at kk.
+        ! therefore, use these particular levels for the input fields, and indicate in 
+        ! the field name within the files
+        call write_var(unit_input,unit_output,'cs_col_in(kk)',cs_col_in(yaml%lev_print))
```
I am getting an error with `cs_col_in(kk)`; Invalid input parameter name: `cs_col_in(kk)`. May you change this name to `cs_col_in_kk`. Same with all variable names with `(`.

brigaudet replied:
OK, these have been so renamed (along with (kp1) to _kp1)

odiazib commented on components/eam/src/chemistry/yaml/cam_ndrop/f90_yaml/update_from_cldn_profile_beg_yml.f90:
```diff
@@ -0,0 +1,172 @@
+
+        call write_var(unit_input,unit_output,'ntot_amode',ntot_amode)
+        call write_var(unit_input,unit_output,'nspec_amode',nspec_amode)
+
```
Please add `cldn_col_in` and `raercol_nsav` at `kp1_yaml`.

brigaudet replied:
OK, these fields have been added.

Commit:
1)  Fix output variable formatting for update_from_newcld and update_from_cldn_profile, so only one cell is present.
2)  Remove '()' from variable names in output files, and add values at kk+1 for additional variables in update_from_cldn_profile.

odiazib reviewed (approved):
It looks great!

brigaudet commented:
![image](https://github.com/eagles-project/e3sm_mam4_refactor/assets/48535869/c6f9ab13-418d-498b-956e-d78a0af6b879)

All longer tests pass, will go ahead and merge.
